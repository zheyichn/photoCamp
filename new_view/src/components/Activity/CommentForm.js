import React from "react";
import { Button, TextField, Box, Grid } from "@mui/material";
import { followButtonStyle } from "../../styles/buttonStyles";

export default function CommentForm(props) {
  return (
    <Grid mt={2}>
      <form>
        <TextField
          className="commentField"
          label="Write your comment here"
          rows={1}
          variant="standard"
          defaultValue=""
          fullWidth
          inputRef={props.newComment}
        />
        <Box display="flex" justifyContent="flex-end" mt={2}>
          <Button
            className="postCommentBtn"
            variant="contained"
            onClick={props.handlePostClick}
            sx={followButtonStyle}
          >
            Post
          </Button>
        </Box>
      </form>
    </Grid>
  );
}
