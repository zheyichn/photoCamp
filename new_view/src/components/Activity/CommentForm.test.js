import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CommentForm from "./CommentForm";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

describe("CommentForm Component", () => {
  const handleClick = jest.fn();
  // Test1 comment display
  test("When user enters a comment it is displayed", async () => {
    render(<CommentForm handlePostClick={handleClick} />);
    // create a reference to the textfield
    const element = screen.getByRole("textbox");
    // type some comments into the textbox
    await userEvent.type(element, "this is a comment");
    // assertion: verify that the text is in the textbox
    expect(element).toHaveValue("this is a comment");
  });

    // Test2: match snapshots
    it("CommentForm renders correctly", () => {
      const tree = renderer
        .create(<CommentForm handlePostClick={handleClick} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});
