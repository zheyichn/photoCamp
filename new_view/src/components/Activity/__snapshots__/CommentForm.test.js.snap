// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm Component CommentForm renders correctly 1`] = `
<div
  className="MuiGrid-root css-1l6dni3-MuiGrid-root"
>
  <form>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Write your comment here
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1a1fmpi-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          defaultValue=""
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          rows={1}
          type="text"
        />
      </div>
    </div>
    <div
      className="MuiBox-root css-iusmz7"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-wuoe3o-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Post
      </button>
    </div>
  </form>
</div>
`;
